File: server/model/books.js
const mongoose = require('mongoose')

const booksSchema = mongoose.Schema(
    {
        name: {
            type: String,
            required: [true, "Please enter a book name"]
        },
        price: {
            type: String,
            required: true,
        },
        image: {
            type: String,
            required: true,
        },
        author: {
            type: String,
            required: true,
        }
    },
    {
        timestamps: true
    }
)


const Book = mongoose.model('Book', booksSchema);

module.exports = Book;

File: lib/models/books.dart
class Book {
  final String id;
  final String title;
  final String author;
  final String imageUrl;
  final String price;

  const Book({
    required this.price,
    required this.id,
    required this.title,
    required this.author,
    required this.imageUrl,
  });

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      imageUrl: json['image'],
      title: json['name'],
      author: json['author'],
      price: json['price'],
      id: json['_id'],
    );
  }
}

List<Book> books = [

];
